### ISSUE TITLE
Legion toolkit doesn't work well with screenreaders

### ISSUE THREAD

**Issue by VIPPotato** [BartoszCichecki/LenovoLegionToolkit#1108]
### Rules

- [X] I made myself familiar with the README.
- [X] I checked closed issues for similar ones and did not find a similar problem mentioned before.
- [X] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.18.0

### OS

Windows 11 23H2 (AMD64) build 22631.2861

### Device

Legion 5 15ITH6H

### BIOS version

H1CN56WW

### What's wrong?

So, I'm blind, and I'm using a software called screenreader [the one I'm using is NVDA]. 
Legion toolkit offers much more features than vantage so I decided to try it out. I can access vantage easily because it's like a web page. 
I'm navigating my computer with a keyboard, and while I can change things easily with tab, arrows and space, I don't hear labels, for example if I focus on the overdrive mode, it just reads it like "toggle button not pressed", but doesn't say what that toggle does. 
I know nothing about coding so can't help in any way, but I'm sure legion toolkit can be easily made accessible for other blind legion users.

### How to reproduce the bug?

Well, I guess get a portable copy of nvda, run it, open legion toolkit, close your eyes and try tabbing around the window.

### What is the behavior that you expected?

I expect all buttons to be fully labelled.

### Logs

[log_2024_01_07_12_47_18.txt](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/13853467/log_2024_01_07_12_47_18.txt)

### Additional information

I think this might help: 
https://learn.microsoft.com/en-us/dotnet/framework/ui-automation/accessibility-best-practices#place-names-titles-and-descriptions-on-ui-objects-frames-and-pages

**Comment by BartoszCichecki** [comment]
Hey, excuse my ignorance, but I am not very familiar with providing good screen reader support. I would like to improve though, so could you give some hints what specifically should be improved (in addition to describing switches and checkboxes)? Or maybe give some good examples of apps that are easy for you to use so I can compare?

**Comment by VIPPotato** [comment]
Hello, 
I'm glad to hear that you're interested in improving accessibility of this app. 
At the moment I can't recall any app written in NET framework that I'm using daily. Most of the apps these days are being made with electron. 
But about what could be improved, for example I'm not able to access that part of the app where it's possible to update drivers, set actions, etc. Also, when there's a new update, I have to download it from github because I'm not able to focus the download button with a keyboard. It would also be cool if these descriptions that are next to each toggle could be narrated atomatically, something like this: "flip to start toggle button  not pressed Turn on the laptop when you open the lid.". For now I'm able to read what each thing does by using a thing called object navigation in my screenreader, which kinda works like moving the mouse and focusing objects, but by using a keyboard. Also, the thing that could be added to improve accessibility is an option in settings to disable real time components usage, like CPU and GPU clocks. My screenreader sees them as progress bars and constantly says the progress of them which is annoying. I really am not sure how else I can help.

**Comment by BartoszCichecki** [comment]
Thanks for the tips. Just installed NVDA and.... well it sucks. I have better idea what is missing now. It will take some time to go through all elements to describe them properly but, I will post a build as soon as I have something for you to test.

**Comment by BartoszCichecki** [comment]
Hey @VIPPotato,

Here is a test build you can download and check out: [LenovoLegionToolkitSetup.zip](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/14089451/LenovoLegionToolkitSetup.zip)

I did my best, but the UI framework I use for some reason doesn't cooperate with NVDA nicely for some element, so some mouse navigation will still be required I am afraid, but I hope what was done will be some improvement.

Regarding the "progress bars", I can't seem to find a way to make NVDA ignore them, but at the bottom of the main page there is a "Customize" button. The first toggle there called Sensors allows to turn them off.

**Comment by BartoszCichecki** [comment]
Sorry, uploaded wrong ZIP file, here is correct one:
[LenovoLegionToolkitSetup.zip](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/14089475/LenovoLegionToolkitSetup.zip)
